src/sys/windows/fdentry.rs

```
@@ -27,12 +24,8 @@ pub struct FdEntry {
 impl Drop for FdObject {
     fn drop(&mut self) {
         if self.needs_close {
-            unsafe {
-                if winapi::um::handleapi::CloseHandle(self.raw_handle) == 0 {
-                    // TODO: use DWORD WINAPI GetLastError(void) to get error
-                    eprintln!("FdObject::drop(): couldn't close raw Handle");
-                }
-            }
+            winx::handle::close(self.raw_handle)
+                .unwrap_or_else(|e| eprintln!("FdObject::drop(): {}", e))
```

FWIW we [explicitly ignore errors in the Rust standard library](https://github.com/rust-lang/rust/blob/53ae6d2eb5352cc20f2e9b92847d92e935639f3c/src/libstd/sys/windows/handle.rs#L60) for errors on `close`. We largely do that to mirror weirdness on Unix, but perhaps this should mirror that?

(or would it be possible to use a `File` from the standard library maybe?)

---

Thanks for the pointer! So the reason we've got error handling here is essentially because I mirrored the implementation here from the corresponding [*nix implementation in wasi-common](https://github.com/CraneStation/wasi-common/blob/master/src/sys/unix/fdentry.rs#L26). Having said that, it might indeed be obsolete :-)

(Hmm, perhaps we could use `File` instead. If we could, then, again, we could consolidate a fair amount of *nix and Win code, so it'd be amazing!)

---

Ah yeah that makes sense (mirroring) and I don't want to try to pile on too much into one PR which is just trying to get things done!